### PR TITLE
added error-handling

### PR DESCRIPTION
**Pull request by nishi-ai**

### CODE REVIEW

**Comment by benevbright** [comment]
Ok, let me explain a bit about two different types of web applications. 

1. Serving web app(front-end) and the backend in one server. 

One server means one computer and one program, here is Node.js. 
Node.js even can serve html to the users. https://stackoverflow.com/a/31056472/949795
When it's the case, you can find `html` or some stuff with naming `render`, they are some clues that that Node.js serves the front end as well. 

So in one project, there is web app (html/JS/CSS) and route for html, and at the same time, there are routes for the API endpoints, like `/registration` for POST method. This approach is sometimes very convenient because you don't have to deploy two different projects. 

But modern web projects are mostly like the following one

2. Deploying web app and server app (backend) separately. (different ports, different url)

In this case, we separate project into two projects, FE and BE, different repos. And more importantly, deploy them separately. 
So FE app build by React has only FE stuff. and we deploy like `www.airbnb.com`. 

and you deploy the Node.js server app in `api.airbnb.com`. and there is no `html` return or rendering stuff. (There is technique called "server side rendering" but we forget about this for now). What this app does it only serving API routes for FE, like `api.airbnb.com/bookings` and handle DB stuff. Nothing about FE.

And FE app calls these endpoints to send requests or get the list of bookings. But again, FE is in `www.airbnb.com` and BE is in `api.airbnb.com` and BE url is not used by the user, it's used by only FE code.


We're the 2 case. So we don't need to return HTML rendering about 500 here. We need to return 500 error to FE, and let FE handle this error code. It's up to them how they render the page with the error code.

File: routes/admin.js [hunk 1]
```diff
@@ -8,4 +8,11 @@ const router = express.Router();
 // /admin/players => GET
 router.get('/players', indexController.getIndex);
 
+router.get('/debug', (req, res, next) => {
```
Comment (benevbright) [inline routes/admin.js:11]: Let's remove this. It was for testing for the moment.

File: controllers/registerController.js [hunk 2]
```diff
@@ -44,7 +44,15 @@ exports.postRegistraionInfo = (req, res, next) => {
         })
         // catching errors related to inserting the document into the database
         .catch(err => {
-            console.log('Error is occured:', err);
-            res.status(500).json({ message: 'Registration for the player failed.' });
+            if (err.code === 11000) {
+                // Duplicate email , catched error before saving into db
+                return res.status(500).json( {
```
Comment (benevbright) [inline controllers/registerController.js:49]: Let's send `409` code. Meaning, "Conflict"

File: controllers/registerController.js [hunk 3]
```diff
@@ -44,7 +44,15 @@ exports.postRegistraionInfo = (req, res, next) => {
...
+                    success: false,
+                    code: 11000,
+                    message: 'This email is already taken'
```
Comment (benevbright) [inline controllers/registerController.js:52]: The error message is better with short form. e.g) `duplicated-record`.

File: controllers/registerController.js [hunk 4]
```diff
@@ -25,6 +25,8 @@ exports.postRegistraionInfo = (req, res, next) => {
         // the collction will be created dynamically if it does not exist yet
         .collection('players')
         .insertOne({
+            // _id for testing duplicated-record
+            _id: "6149cebfe78b085815277716",
```
Comment (benevbright) [inline controllers/registerController.js:29]: good. 🙌🙌 but remove :)

Comment (nishi-ai) [reply controllers/registerController.js:29]: can i remove it after implementing the error message in FE?

**Review by benevbright — APPROVED** [review]
Great work!!

**Pull request merged**